File: jute_mark_india/jute_mark_india/web_form/jute_mark_india_registration/jute_mark_india_registration.js
frappe.ready(function() {
  frappe.web_form.on('state', (field, value) =>{
    frappe.web_form.set_value("district", );
    frappe.web_form.set_value("tahsil__taluka", );
    frappe.web_form.set_value("pin_code", );
    frappe.call({
      method: 'jute_mark_india.jute_mark_india.web_form.jute_mark_india_registration.jute_mark_india_registration.get_districts',
      args: {
        state: value
      },
      freeze: true,
      callback: (r) => {
        var options = [];
        for (var i = 0; i < r.message.length; i++) {
          options.push({
            'label': r.message[i].name,
            'value': r.message[i].name
          });
        }
        var field = frappe.web_form.get_field("district");
        field._data = options;
        field.refresh();
      }
    });
  });

  frappe.web_form.on('district', (field, value) =>{
    frappe.call({
      method: 'jute_mark_india.jute_mark_india.web_form.jute_mark_india_registration.jute_mark_india_registration.get_tahsil_taluka',
      args: {
        district: value
      },
      freeze: true,
      callback: (r) => {
        var options = [];
        for (var i = 0; i < r.message.length; i++) {
          options.push({
            'label': r.message[i].name,
            'value': r.message[i].name
          });
        }
        var field = frappe.web_form.get_field("tahsil__taluka");
        field._data = options;
        field.refresh();
      }
    });
    frappe.call({
      method: 'jute_mark_india.jute_mark_india.web_form.jute_mark_india_registration.jute_mark_india_registration.get_pincodes',
      args: {
        district: value
      },
      freeze: true,
      callback: (r) => {
        var options = [];
        for (var i = 0; i < r.message.length; i++) {
          options.push({
            'label': r.message[i].name,
            'value': r.message[i].name
          });
        }
        var field = frappe.web_form.get_field("pin_code");
        field._data = options;
        field.refresh();
      }
    });
  });

  frappe.web_form.on('mobile_number', (field, value) =>{
    if(value){
      if(isNaN(value)){
        field.set_value('');
      }
      if(value.length>10) {
        frappe.msgprint('Phone Number Must be 10 Digits');
        field.set_value('');
      }
      frappe.call({
        method: 'jute_mark_india.jute_mark_india.web_form.jute_mark_india_registration.jute_mark_india_registration.validate_mobile_number',
        args: {
          mobile_no: value
        }
      });
    }
  });

  frappe.web_form.on('email_id', (field, value) =>{
    if(value){
      frappe.call({
        method: 'jute_mark_india.jute_mark_india.web_form.jute_mark_india_registration.jute_mark_india_registration.validate_email',
        args: {
          email_id: value
        }
      });
    }
  });

  frappe.web_form.on('aadhar_number', (field, value) =>{
    if(value){
      if(isNaN(value)){
        field.set_value('');
  		}
      if(value.length>12) {
        frappe.msgprint('Aadhar Number Must be 12 Digits');
        field.set_value('');
      }
    }
  });
});

frappe.web_form.validate = () =>{
    let udyog_aadhar = frappe.get_form.get_value('udyog_aadhar');
    var pattern = /^[A-Z]{6}-[A-Z]{2}-[0-9]{2}-[0-9]{7}$/
    if(!udyog_aadhar.match(pattern)){
      frappe.throw('Enter Valid Udayam Aadhar');
      return False;
    }
}

File: jute_mark_india/jute_mark_india/web_form/jute_mark_india_registration/jute_mark_india_registration.py
import frappe
import re

def get_context(context):
	pass

@frappe.whitelist(allow_guest=True)
def get_districts(state):
	if state:
		districts = frappe.db.get_all('District', filters={'state':state})
	else:
		districts = frappe.db.get_all('District')
	return districts

@frappe.whitelist(allow_guest=True)
def get_tahsil_taluka(district):
	if district:
		tahsil_talukas = frappe.db.get_all('TalukaTahsil', filters={'district':district})
	else:
		tahsil_talukas = frappe.db.get_all('TalukaTahsil')
	return tahsil_talukas

@frappe.whitelist(allow_guest=True)
def get_pincodes(district):
	if district:
		pincodes = frappe.db.get_all('Pincode', filters={'district':district})
	else:
		pincodes = frappe.db.get_all('Pincode')
	return pincodes

@frappe.whitelist(allow_guest=True)
def validate_mobile_number(mobile_no):
	if frappe.db.exists("Jute Mark India Registration form",{'mobile_number':mobile_no}):
		frappe.throw("Registration Form with this Mobile Number Already Exists")

@frappe.whitelist(allow_guest=True)
def validate_email(email_id):
	if frappe.db.exists("Jute Mark India Registration form",{'email_id':email_id}):
		frappe.throw("Registration Form with this Email Id Already Exists")


@frappe.whitelist(allow_guest=True)
def get_last_signup_data():
	'''
		Method to get details of last signup OTP
	'''
	if frappe.db.exists('User Registration', { 'verified':1 }):
		email_id, mobile_no = frappe.db.get_value('User Registration', { 'verified':1 }, ['email_id', 'mobile_number'])
		return { 'email_id':email_id, 'mobile_no':mobile_no }
	else:
		return 0
